app/src/main/java/io/github/louistsaitszho/lineage/activities/MainActivity.kt

```diff
@@ -2,12 +2,113 @@ package io.github.louistsaitszho.lineage.activities
+                runAdapter()
+                return true
+            }
+            R.id.large -> {
+                layoutOption = LARGE
+                runAdapter()
+                return true
+            }
+            else -> return false
+        }
+
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        //Default layoutOption
+        layoutOption = NO_THUMBNAIL
+
+        //Method that Instantiates and runs the adapter
+        runAdapter()
+    }
+
+
+    fun runAdapter(){
+
+        //POOR IMPLEMENTATION DUE TO MY POOR KOTLIN SKILLS
+        var someSome: ItemListUnit = ItemListUnit("January","http://flaglane.com/download/german-flag/german-flag-large.png","This is the URL 2")
+        var someSome1: ItemListUnit = ItemListUnit("February ","http://flaglane.com/download/german-flag/german-flag-large.png","This is the URL 2")
+        var someSome2: ItemListUnit = ItemListUnit("March","https://upload.wikimedia.org/wikipedia/commons/thumb/5/5b/Flag_of_Hong_Kong.svg/2000px-Flag_of_Hong_Kong.svg.png","This is the URL 2")
+        var someSome3: ItemListUnit = ItemListUnit("May","http://www.freepngimg.com/download/india/4-2-india-flag-png-hd.png","This is the URL 2")
+        var someSome4: ItemListUnit = ItemListUnit("June","https://upload.wikimedia.org/wikipedia/commons/1/17/Flag_of_Mexico.png","This is the URL 2")
+        var someSome5: ItemListUnit = ItemListUnit("July","https://drive.google.com/open?id=0BzUGskuag7oNZHR2TFNsbnFxaU0","This is the URL 2")
+        var someSome6: ItemListUnit = ItemListUnit("August","https://drive.google.com/open?id=0BzUGskuag7oNZHR2TFNsbnFxaU0","This is the URL 2")
+        var someSome7: ItemListUnit = ItemListUnit("September","https://drive.google.com/open?id=0BzUGskuag7oNZHR2TFNsbnFxaU0","This is the URL 2")
+        var someSome8: ItemListUnit = ItemListUnit("October","https://drive.google.com/open?id=0BzUGskuag7oNZHR2TFNsbnFxaU0","This is the URL 2")
+        var someSome9: ItemListUnit = ItemListUnit("Noviembre","https://drive.google.com/open?id=0BzUGskuag7oNZHR2TFNsbnFxaU0","This is the URL 2")
+        var someSome10: ItemListUnit = ItemListUnit("Diciembre","https://drive.google.com/open?id=0BzUGskuag7oNZHR2TFNsbnFxaU0","This is the URL 2")
+        var someSome11: ItemListUnit = ItemListUnit("THIS IS A VERY LONG STRING TO CHECK HOW LONG TEXT BEHAVES IN THIS SITUATION. IT'S ALSO FRIDAY TODAY, RAINY DAY","http://flaglane.com/download/german-flag/german-flag-large.png","This is the URL 2")
+
+        //List that will hold the dummy information
+        listItems = mutableListOf(someSome, someSome1,someSome2,someSome3,someSome4,someSome5,someSome6, someSome7,someSome8,someSome9,someSome10,someSome11)
+
+
+        //Instantiating and setting recycler view attributes
+        recyclerView = findViewById(R.id.recycler_view)
+        (recyclerView)!!.setHasFixedSize(true)
```

Comment:
in kotlin you should avoid using !! as much as possible: https://kotlinlang.org/docs/reference/null-safety.html 
Since everything is design to be non-null, if you declare something is null-able, you should also think about what the behaviour should be if it is null. In this case, if the reyclerview cannot be found, just do nothing, and you should use ? instead

Reply:
Corrected.

recyclerView is no longer used.